### PR TITLE
Support user metadata in child workflow, timer, and activity

### LINKED ISSUE #830: [Feature Request] Support user metadata in child workflow, timer, and activity
### Describe the solution you'd like

See https://github.com/temporalio/features/issues/486. Specifically:

* Accept "static summary" and "static details" optional strings on child workflow and populate user metadata with them
* Accept "summary" on timers and populate user metadata with it
* Accept "summary" on activities and populate user metadata with it
* Mark experimental

### PR DESCRIPTION
**Pull request by yuandrew**
## What was changed

added `static_summary` and `static_details` to  child workflow

added `summary` to timer and activity

and plumbed all new fields to user_metadata in command sent to server. 

## Why?

This allows individual SDK languages to pass this information to the server, allowing it to show up in the UI.

## Checklist

1. Closes #830

2. How was this tested:
Wrote new tests to validate metadata is being passed to server

### CODE REVIEW

**Review by cretz — APPROVED** [review]
This looks great! May want to wait for @Sushisource to approve before merging.

File: core/src/worker/workflow/machines/child_workflow_state_machine.rs
Comment (Sushisource) [inline core/src/worker/workflow/machines/child_workflow_state_machine.rs:457]: This is a little nitpicky, but, instead of cloning here, since these values aren't used by `start_child_workflow_cmd_to_api`, let's instead https://doc.rust-lang.org/std/mem/fn.take.html the string out and we avoid a clone

File: core/src/worker/workflow/machines/timer_state_machine.rs
Comment (yuandrew) [reply core/src/worker/workflow/machines/timer_state_machine.rs:86]: Without the clone, I get the error 
```
cannot move out of a shared reference
move occurs because value has type `temporal_sdk_core_protos::coresdk::workflow_commands::StartTimer`, which does not implement the `Copy` traitrustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic message [0]?0#file:///Users/andrewyuan/temporal/core/core/src/worker/workflow/machines/timer_state_machine.rs)
```

Looks like we clone the attrs of Activity https://github.com/yuandrew/sdk-core/blob/dd37e2946a385090ded0fc3312180027002d4b16/core/src/worker/workflow/machines/activity_state_machine.rs#L134

Comment (yuandrew) [reply core/src/worker/workflow/machines/timer_state_machine.rs:86]: hmm, curious why we're even getting this error in the first place.. the line I added for `user_metadata` doesn't involve `s` directly, but maybe because I'm cloning `attribs.summary`, it messes with the borrow checker, when s is created?`let mut s = Self::new(attribs);`

Is there a different workaround here, or would we have to clone (or implement `Copy`) here?

Comment (Sushisource) [reply core/src/worker/workflow/machines/timer_state_machine.rs:86]: Ah, yeah I didn't see it getting moved later. This clone is fine to keep then, but we could change the line above to also `take` the summary. 

And yeah, in general I want to reduce how many places lots of various strings get cloned, like the place you linked, but that's a bigger project. We may not ever do it since ultimately it's not that big of a deal but, I want to avoid adding more spots where we unnecessarily clone.

**Pull request merged**
